node-api-postgres/queries.js

```diff
@@ -24,9 +24,30 @@ const checkSites = (request, response) => {
     if (error) {
       throw error;
     }
-    response.status(200).json(results.rows);
+    // for all sites, for all queries return status
+    // axios.all is returning promise collecting all responses
+    // Promise.all is creating promise fulfilled when all "child" promises are fulfilled
+    Promise.all(results.rows.map(
+        ({name, urls}) => axios.all(
+            urls.map(url => axios.get(url).then(response => [name, url, response.status])))
+        )
+    ).then(results => {
+      // when all queries to external servers are finished
+
+      // build response like: {name: {url: status}}
+      const flat = results.reduce((acc, list) => [...list]).reduce(
+          (acc, [name, url, status]) => {
+            const current = acc[name] || {};
+            current[url] = status;
+            acc[name] = current;
+            return acc;
+          }, {}
+      );
+      response.status(200).json(flat);
+    });
```

Comment:
answers from external services can take a long time. It's best to try to do all of them as promises and wait for the whole bunch.

If there are 10 urls and each take ~.5s doing them one by one will take ~5s. but if it's done all together it should take less than 1s.